File: milestones/milestone-8-react-native-fundamentals/react-native-app/modules/native-demo-module/android/src/main/java/expo/modules/nativedemomodule/NativeDemoModule.kt
package expo.modules.nativedemomodule

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class NativeDemoModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("NativeDemoModule")

    AsyncFunction("getNativeDemoInfo") {
      mapOf(
        "platform" to "android",
        "message" to "Hello from the Android Expo Module demo."
      )
    }
  }
}

File: milestones/milestone-8-react-native-fundamentals/react-native-app/modules/native-demo-module/ios/NativeDemoModule.swift
import ExpoModulesCore

public final class NativeDemoModule: Module {
  public func definition() -> ModuleDefinition {
    Name("NativeDemoModule")

    AsyncFunction("getNativeDemoInfo") {
      [
        "platform": "ios",
        "message": "Hello from the iOS Expo Module demo."
      ]
    }
  }
}
